src/main/scala/com/mwz/sonar/scala/GlobalConfig.scala

```diff
@@ -93,6 +93,7 @@ final class GlobalConfig(config: Configuration) {
         config.getAs[String](PR_NUMBER),
         ConfigError(s"Please provide a pull request number ($PR_NUMBER).")
       )
+      githubApiUrl = config.getAs[String](PR_GITHUB_API_URL).getOrElse(DEFAULT_GITHUB_API_URL)
```

Comment:
I believe you may get this value directly as an **URI**, instead of as an **String**.

Reply:
@BalmungSan 
Thanks for your comments.
I think it seems to be to create a getAs [T <: Uri] method in SonarConfig.
However, if i do, many existing codes (ex. Val prUri: String> val val prUri: Uri, etc.) are likely to be affected. Is this an okay?

Reply:
Oh right I got confused and for a moment I thought we were using **pureconfig** which does has that method.

However, you may want to give a look to how ` val baseUrl` is parsed.

Reply:
yeah, I agree that the url should be parsed otherwise it could easily blow up later when you're calling `Uri.unsafeFromString`, as the name suggests :)

You could do something like that:
```scala
githubApiUrl <- EitherT.fromEither(
  config
    .getAs[String](PR_GITHUB_API_URL)
    .map(Uri.fromString)
    .fold[ConfigErrorOr[Uri]](Right(DEFAULT_GITHUB_API_URL))(_.leftMap(f => ConfigError(f.sanitized)))
)
```

This is slightly different from the `baseUrl`, because we can accept a missing value (and default to `DEFAULT_GITHUB_API_URL`), but we want to fail early when there's a parsing error.

You'll also need to change the `DEFAULT_GITHUB_API_URL` to `Uri.uri("https://api.github.com")` and you should be on the safe side.

Reply:
I've modified it. Please make sure that it's the right one.